### PR TITLE
MAINT: make dual_annealing signature match other optimizers

### PR DESCRIPTION
**Pull request by sgubianpm**
PR for aligning dual_annealing interface.

### CODE REVIEW

**Comment by ilayn** [comment]
This looks like it is breaking the backwards compatibility.

**Comment by andyfaff** [comment]
@ilayn, the function is public, but it hasn't been in a scipy release yet

**Comment by ilayn** [comment]
Ah OK. My bad sorry

**Comment by ilayn** [comment]
I'm a bit confused is this already in 1.2? Do we need to backport this? Because I think this is in for version 1.2 already right?

**Comment by sgubianpm** [comment]
Done. Adjusted also the unit tests and documentation.

File: scipy/optimize/_dual_annealing.py [hunk 1]
```diff
@@ -495,6 +491,8 @@ def dual_annealing(func, x0, bounds, args=(), maxiter=1000,
             - 2: detection done in the dual annealing process.
 
         If the callback implementation returns True, the algorithm will stop.
+    x0 : ndarray, shape(n,), optional
+        A single initial starting point coordinates.
```
Comment (tylerjereddy) [inline scipy/optimize/_dual_annealing.py]: Plural "coordinates" doesn't sound right here?

**Comment by tylerjereddy** [comment]
@ilayn It is in the `1.2.0rc1` release *candidate*, but it looks like we'll want to backport this when it is ready based on discussion in #9547 before the actual final release.

File: scipy/optimize/_dual_annealing.py (hunk 1 above)
Comment (sgubianpm) [reply scipy/optimize/_dual_annealing.py]: @tylerjereddy, I was meaning coordinates of a single starting point (like a n-dimension starting point), do you think it has to be "coordinate". I am not a native English speaker, so I may wrote this wrongly...

Comment (tylerjereddy) [reply scipy/optimize/_dual_annealing.py]: Maybe something similar to your suggestion then -- "Coordinates of a single n-dimensional starting point"?

Comment (sgubianpm) [reply scipy/optimize/_dual_annealing.py]: Thanks a lot! Sounds good. Adjusted.

**Comment by rgommers** [comment]
LGTM, merged. Thanks @sgubianpm

**Pull request merged**